### PR TITLE
Fix docker image not pulled with --image option

### PR DESCRIPTION
**Pull request by edingroot**
In paasta spark-run, since the Docker image digest used to launch the Spark cluster is obtained by inspecting local Docker images, we need to ensure that the Docker image exists locally or is pulled in all scenarios.

```bash
aws-okta -r core-ml -a dev -k -p dev
docker rmi docker-dev.yelpcorp.com/ad-targeting-batches-austinzh:0.117.8
paasta spark-run --image docker-dev.yelpcorp.com/ad-targeting-batches-austinzh:0.117.8 --aws-profile=dev --cmd bash
```

Before:
image

After:
image

### CODE REVIEW

File: paasta_tools/cli/cmds/spark_run.py [hunk 1]
```diff
@@ -513,25 +513,33 @@ def get_docker_run_cmd(
...
     if args.image:
-        return args.image
+        docker_url = args.image
+    else:
+        try:
+            docker_url = instance_config.get_docker_url()
+        except NoDockerImageError:
+            print(
+                PaastaColors.red(
+                    "Error: No sha has been marked for deployment for the %s deploy group.\n"
```
Comment (88manpreet) [inline paasta_tools/cli/cmds/spark_run.py:533]: Nit: Is PaastaColors compatible with f-strings? If so, can we use them for consistency?

File: paasta_tools/cli/cmds/spark_run.py [hunk 2]
```diff
@@ -513,25 +513,33 @@ def get_docker_run_cmd(
...
-            PaastaColors.red(
-                "Error: No sha has been marked for deployment for the %s deploy group.\n"
-                "Please ensure this service has either run through a jenkins pipeline "
-                "or paasta mark-for-deployment has been run for %s\n"
-                % (instance_config.get_deploy_group(), args.service)
-            ),
-            sep="",
-            file=sys.stderr,
-        )
-        return None
```
Comment (88manpreet) [inline paasta_tools/cli/cmds/spark_run.py:534]: In Line548, https://github.com/Yelp/paasta/pull/3614/files#diff-30f8e9c33224158fab1e77efc4004f7d2344230ea7b4ffbf64b3fca21513cc00R548 should we be checking if docker image exists locally befor pulling? Or the check is happening implicitly?

**Review by 88manpreet — APPROVED** [review]
Left one comment, but lgtm otherwise.

File: paasta_tools/cli/cmds/spark_run.py (hunk 2 above)
Comment (edingroot) [reply paasta_tools/cli/cmds/spark_run.py:534]: For the case of the local image is newer than the one on registry (built, tagged locally but didn't push), `docker pull` will always pull the latest version on the registry anyway.

More details:
https://github.com/moby/moby/issues/10291#issuecomment-71655034

**Pull request merged**